src/http-client.ts

```diff
@@ -0,0 +1,77 @@
+import * as superagent from 'superagent';
+const HTTPSAgent = require('agentkeepalive').HttpsAgent;
+const HTTPAgent = require('agentkeepalive');
+
+type SuperAgentStatic = typeof superagent;
+type KeepAliveOptions = {
+  timeout: number;
+  freeSocketTimeout: number;
+};
+
+const DEFAULT_SOCKET_TIMEOUT = 60000;
+/**
+ * Factory for creating a singleton superagent instance with connection pooling
+ */
+class SuperagentFactory {
+  instance: SuperAgentStatic | null;
+  private keepaliveOptions: KeepAliveOptions;
+  private httpAgent;
+  private httpsAgent;
+  
+  constructor() {
+    this.instance = null;
+    
+    this.keepaliveOptions = {
```

Comment:
Should be named agentOptions since it's the agent options and not specifically for keep alive